### PR TITLE
vim-patch:8.1.1084: cannot delete a match from another window

### LINKED ISSUE #4178: matchdelete call fails when current window is not window associated with id
`matchaddpos` has the following signature:

```
matchaddpos({group}, {pos} [, {priority} [, {id} [, {dict}]]])
```

This allows me to add a match in a specific window because the `dict` argument is defined as follows:

```
                The optional {dict} argument allows for further custom
                values. Currently this is used to specify a match specific
                conceal character that will be shown for hl-Conceal
                highlighted matches. The dict can have the following members:

                        conceal     Special character to show instead of the
                                    match (only for hl-Conceal highlighted
                                    matches, see :syn-cchar)
                        window      Instead of the current window use the
                                    window with this number or window ID.
```

The value returned from `matchaddpos` is an ID which I had assumed was independent of window. But calling `matchdelete` with a current window of W2, say, passing in an ID that is the result of a call to `matchaddpos` for window W1, fails with:

```
E803: ID not found: <ID>
```

Am I doing something wrong here?

### LINKED ISSUE #12110: `matchdelete` should allow passing in window number
NVIM v0.5.0-dev

`matchdelete` doesn't allow passing in window number, which makes the window number passed into `matchadd` useless.

### PR DESCRIPTION
**Pull request by Shougo**
Problem:    Cannot delete a match from another window. (Paul Jolly)
Solution:   Add window ID argument to matchdelete(), clearmatches(),
            getmatches() and setmatches(). (Andy Massimino, closes vim/vim#4178)
https://github.com/vim/vim/commit/aff749145e23c0f20b5158d1d3a942948ed138e3

Fix https://github.com/neovim/neovim/issues/12110

### CODE REVIEW

File: src/nvim/eval/funcs.c [hunk 1]
```diff
@@ -952,12 +953,30 @@ static void f_cindent(typval_T *argvars, typval_T *rettv, FunPtr fptr)
...
  * "clearmatches()" function
  */
 static void f_clearmatches(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 {
-  clear_matches(curwin);
+  win_T *win = get_optional_window(argvars, 0);
+
+  if (win != NULL) {
+    clear_matches(curwin);
+  }
```
Comment (teto) [inline src/nvim/eval/funcs.c:979]: `shouldn't it be clear_matches(win)` ?

**Review by teto — CHANGES_REQUESTED** [review]

File: src/nvim/eval/funcs.c (hunk 1 above)
Comment (Shougo) [reply src/nvim/eval/funcs.c:979]: Thanks.  Fixed.

**Comment by Shougo** [comment]
I don't know why `builds.sr.ht: openbsd.yml `fails.

**Comment by Shougo** [comment]
Almost tests are passed.

File: src/nvim/eval/funcs.c (hunk 1 above)
Comment (teto) [reply src/nvim/eval/funcs.c:979]: the vim patch had the same issue so maybe there is a follow up patch that fixes it and other things.

**Comment by teto** [comment]
openbsd CI is broken at the moment but the rest should pass

**Comment by Shougo** [comment]
> openbsd CI is broken at the moment but the rest should pass

OK.  It seems passed.

**Comment by teto** [comment]
CC @janlazo since https://github.com/neovim/neovim/pull/11637 mentioned it

**Pull request merged**